Neos.Neos/Classes/Fusion/Helper/DimensionHelper.php

```diff
@@ -0,0 +1,202 @@
+
+use Neos\ContentRepository\Core\Dimension\ContentDimension;
+use Neos\ContentRepository\Core\Dimension\ContentDimensionId;
+use Neos\ContentRepository\Core\Dimension\ContentDimensionValue;
+use Neos\ContentRepository\Core\Dimension\ContentDimensionValues;
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+use Neos\ContentRepository\Core\Projection\ContentGraph\Nodes;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\Eel\ProtectedContextAwareInterface;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\ObjectManagement\DependencyInjection\DependencyProxy;
+
+/**
+ * Helper for nodes in various content dimensions.
+ */
+class DimensionHelper implements ProtectedContextAwareInterface
+{
+    /**
+     * @var ContentRepositoryRegistry
+     */
+    #[Flow\Inject(lazy: true)]
+    protected $contentRepositoryRegistry;
+
+    /**
+     * Get current dimension value for `node`.
+     *
+     * Example::
+     *
+     *     Neos.Dimension.currentValue(node, 'language')
+     *
+     * @param Node $node
+     * @param string|ContentDimensionId $dimensionName
+     * @return string|null
+     */
+    public function currentValue(Node $node, string|ContentDimensionId $dimensionName): ?string
+    {
+        $contentDimensionId = is_string($dimensionName) ? new ContentDimensionId($dimensionName) : $dimensionName;
+        return $node->subgraphIdentity->dimensionSpacePoint->getCoordinate($contentDimensionId);
+    }
+
+    /**
+     * Get original dimension value for `node`. Differs form current value in cases of dimension fallback.
+     *
+     * Example::
+     *
+     *     Neos.Dimension.originValue(node, 'language')
+     *
+     * @param Node $node
+     * @param string|ContentDimensionId $dimensionName
+     * @return string|null
+     */
+    public function originValue(Node $node, string|ContentDimensionId $dimensionName): ?string
+    {
+        $contentDimensionId = is_string($dimensionName) ? new ContentDimensionId($dimensionName) : $dimensionName;
+        return $node->originDimensionSpacePoint->getCoordinate($contentDimensionId);
+    }
+
+    /**
+     * Get default dimension value for the content repository defined by `node`.
+     *
+     * Example::
+     *
+     *     Neos.Dimension.findDefaultValue(node, 'language')
+     *
+     * @param Node $node
+     * @param string|ContentDimensionId $dimensionName
+     * @return string|null
+     */
+    public function findDefaultValue(Node $node, string|ContentDimensionId $dimensionName): ?string
+    {
+        $contentDimensionId = is_string($dimensionName) ? new ContentDimensionId($dimensionName) : $dimensionName;
+        $contentRepository = $this->contentRepositoryRegistry->get($node->subgraphIdentity->contentRepositoryId);
+        $rootValues = $contentRepository
+            ->getContentDimensionSource()
+            ->getDimension($contentDimensionId)
+            ?->getRootValues() ?: [];
+
+        $dimensionValue = reset($rootValues);
```

Comment:
hm, this is not totally correct. Where do we need this? Would be better to read it from Sites comfiguration https://docs.neos.io/guide/manual/site-configuration#site-specific-configuration-reference

the above is ambiguous in case you do not have a single "root" dimension value.

Reply:
@skurfuerst Not entirely sure what to consider right. We decided the first of the root values being the closest thing to a default for now.

Using site configuration can be misleading in cases where the node we want to get informations about could be from a non/site path or totally different cr. Think about a taxon.

What about being more explicit and provide `findDefaultValueByNode(node, dimension)` / `findDefaultValueBySite(site, dimension)`

Could apply to other methods as well where we often pass the node to identify the cr which may not be from the current site.

Reply:
=> neos.Seo

Reply:
ToDo: Remove from helper and solve the edge use case in `Neos.Seo`